scripts/konflux/subscription-manager/subscription-manager-bro.sh

```diff
@@ -0,0 +1,308 @@
+#!/usr/bin/env bash
+
+# This script is for registering container with Red Hat subscription-manager during Konflux builds for getting access
+# to RHEL RPMs during the build.
+# The script was created as a workaround in absence of better options.
+# TODO(ROX-20651): remove this script and switch to use content sets once available.
+
+set -euo pipefail
+
+SCRIPT_NAME="$(basename -- "${BASH_SOURCE[0]}")"
+SCRIPT_DIR="$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )"
+
+KONFLUX_SECRET_NAME="subscription-manager-activation-key"
+SECRET_KEY="activation-key"
+SECRET_PATH="${SCRIPT_DIR}/${SECRET_KEY}"
+SECRET_INFO_URL='https://docs.engineering.redhat.com/pages/viewpage.action?pageId=407312060'
+
+RED_HAT_ORG_ID="11009103"
+TARGETS_LIST_FILE="/tmp/subscription-manager-bro-targets"
+
+# These were figured experimentally with the help of self-test subcommand.
+TARGET_BACKUP_PATHS=(
+    etc/pki/consumer
+    etc/pki/entitlement
+    etc/pki/product
+    etc/pki/product-default
+    etc/yum.repos.d
+    var/lib/rhsm
+    var/cache/ldconfig
+)
+
+
+function main {
+    if [[ "$#" == "0" ]] ; then
+        >&2 echo "Error: command is missing. See the usage below."
+        usage
+        exit 2
+    fi
+
+    local cmd="$1"
+    shift
+
+    local fn
+
+    case "$cmd" in
+    "help" | "--help" | "-h")
+        fn=usage ;;
+    "smuggle")
+        fn=smuggle ;;
+    "register")
+        fn=register ;;
+    "cleanup")
+        fn=cleanup ;;
+    "self-test")
+        fn=self_test ;;
+    "diff")
+        fn=assert_diff ;;
+    *)
+        >&2 echo "Error: unknown command '$1'; call '$SCRIPT_NAME help' to see the usage."
+        exit 2
+        ;;
+    esac
+
+    if [[ "$#" -gt "0" && "$cmd" != "diff" && "$cmd" != "register" ]]; then
+        >&2 echo "Error: too many arguments; call '$SCRIPT_NAME help' to see the usage."
+        exit 2
+    fi
+
+    "$fn" "$@"
+}
+
+function usage {
+    local example_target_dir="/mnt"
+
+    echo "Usage: $SCRIPT_NAME smuggle|register|cleanup|self-test"
+    echo
+    echo "This script enables access to RHEL RPMs during Konflux builds. The intended usage is as follows."
+    echo
+
+    echo -n "1. Make sure there is a '$KONFLUX_SECRET_NAME' secret in Konflux with key name '$SECRET_KEY' and "
+    echo "the actual activation key as a value."
+    echo "   Find where to get the secret from ${SECRET_INFO_URL}"
+
+    echo -n "2. In a Tekton pipeline step before the container build, copy the subscription manager activation "
+    echo "key secret to the source workspace. Use:"
+    echo "   \$ <source-workspace>/$SCRIPT_NAME smuggle"
+    echo -n "   This expects the '$KONFLUX_SECRET_NAME' secret to be mounted as a workspace with the same name "
+    echo "('$KONFLUX_SECRET_NAME')."
```

Comment:
```suggestion
    echo "('/workspace/$KONFLUX_SECRET_NAME')."
```

You're making the assumption that the file is mounted at `/workspace` in https://github.com/stackrox/stackrox/pull/9851/files#diff-1f3958364a018484c00d1f0b65f7418cf78067a7a2d2a77553e8ffa88886548eR120, this may make this comment clearer.

Reply:
Done.